### PR TITLE
feat(suite): add Tabs component

### PR DESCRIPTION
**Pull request by seibei-iguchi**
- added Tabs component
- updated transaction details modal to use Tabs component
- updated subpage navigation to use Tabs component

### Before
https://github.com/user-attachments/assets/48c26220-5671-419c-a718-f11e69e8f07f
Screenshot 2024-11-25 at 10 20 11

### After
https://github.com/user-attachments/assets/a7e4a920-9186-4109-bf2f-367c716d9463
Screenshot 2024-11-25 at 10 21 21

### CODE REVIEW

File: packages/components/src/components/Tabs/Tabs.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import { Tabs as TabsComponent, TabsProps } from './Tabs';
+import { tabsSizes } from './types';
+import { Column } from '../Flex/Flex';
+
+const meta: Meta = {
+    title: 'Tabs',
+} as Meta;
+export default meta;
+
+const TabsApp = (props: Partial<TabsProps>) => {
```
Comment (jvaclavik) [inline packages/components/src/components/Tabs/Tabs.stories.tsx:16]: I know it doesnt matter here, but "app" sounds weird and inconsistent.

**Review by jvaclavik — APPROVED** [review]
Well done!

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/AdvancedTxDetails/AdvancedTxDetails.tsx [hunk 2]
```diff
@@ -81,46 +62,35 @@ export const AdvancedTxDetails = ({
...
-                        $isSelected={selectedTab === 'data'}
-                        onClick={() => setSelectedTab('data')}
-                    >
-                        <Translation id="TR_DATA" />
-                    </TabButton>
-                )}
-            </Row>
-            <Divider margin={{ top: 0, bottom: spacings.md }} />
+            <Tabs
+                items={[
```
Comment (peter-sanderson) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/AdvancedTxDetails/AdvancedTxDetails.tsx]: I strongly prefer to have declarative type: 

```ts
const items: TabItems<TabID> = [.....]
```

This way you do not have to do `as TabID` multiple times

File: packages/components/src/components/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    const updateIndicator = useCallback(() => {
+        if (!activeItemId) return;
+
+        const activeItemEl = tabsRefs.current.get(activeItemId);
+        const width = activeItemEl?.getBoundingClientRect()?.width;
+        const position = activeItemEl?.offsetLeft;
+
+        setIndicatorWidth(width ?? 0);
+        setIndicatorPosition(position ?? 0);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (peter-sanderson) [inline packages/components/src/components/Tabs/Tabs.tsx]: Here I suggest to specify what is missing and why

Comment (seibei-iguchi) [reply packages/components/src/components/Tabs/Tabs.tsx]: It's the opposite, nothing is missing - items is there even though they don't have to.

Comment (peter-sanderson) [reply packages/components/src/components/Tabs/Tabs.tsx]: Ok, so then lets leave the comment about them and why they are needed

**Review by peter-sanderson — APPROVED** [review]

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/AdvancedTxDetails/AdvancedTxDetails.tsx [hunk 4]
```diff
@@ -62,35 +62,26 @@ export const AdvancedTxDetails = ({
...
-                        id: 'data' as TabID,
-                        callback: () => setSelectedTab('data'),
-                        title: <Translation id="TR_DATA" />,
-                        isHidden: network.networkType !== 'ethereum' || !tx.ethereumSpecific,
-                    },
-                ]}
-                activeItemId={selectedTab}
-                margin={{ bottom: spacings.md }}
-            />
+            <Tabs activeItemId={selectedTab} margin={{ bottom: spacings.md }}>
```
Comment (jvaclavik) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/AdvancedTxDetails/AdvancedTxDetails.tsx:65]: `activeItemId` seems to me a bit out of context, what about `activeTabId`?

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/AdvancedTxDetails/AdvancedTxDetails.tsx [hunk 5]
```diff
@@ -62,35 +62,26 @@ export const AdvancedTxDetails = ({
...
+                        <Translation id="TR_INPUTS_OUTPUTS" />
+                    </Tabs.Item>
+                )}
+                {chainedTxs && (
+                    <Tabs.Item id="chained" onClick={() => setSelectedTab('chained')}>
+                        <Translation id="TR_CHAINED_TXS" />
+                    </Tabs.Item>
+                )}
+                {network.networkType === 'ethereum' && tx.ethereumSpecific && (
+                    <Tabs.Item id="data" onClick={() => setSelectedTab('data')}>
```
Comment (jvaclavik) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/AdvancedTxDetails/AdvancedTxDetails.tsx:80]: what about rename it to `setActiveTab` for better consistency?

**Review by jvaclavik — APPROVED** [review]

**Pull request merged**